### DISCUSSION TITLE
Component tests not running after trying to `installGlobals()`

### DISCUSSION THREAD

**Discussion by powerfulech0** [remix-run/remix#8424] [Q&A]
New to Remix, was wondering what I am doing wrong in this approach:

I have a test for a component that uses the fetcher in it:
```
test('loads Program Selector', async () => {
  function TestProgramSelector() {
    return (
      <ProgramSelector
        programs={testPrograms}
        activePrograms={activeTestPrograms}
      />
    );
  }

  const RemixStub = createRemixStub([
    {
      path: '/',
      Component: TestProgramSelector,
      loader() {
        return null;
      },
    },
  ]);

  render(<RemixStub />);

  await waitFor(() => screen.findByText('Program Selector'));
});
```

When I run the tests it gives this error back:

![Screenshot 2024-01-03 at 1 50 19 PM](https://github.com/remix-run/remix/assets/13987197/e841f034-abe3-4d31-8fe6-772abf04b160)

Which led me to trying to install the polyfills with `installGlobals` in a Jest setup file:

```
import { installGlobals } from '@remix-run/node';

beforeAll(() => {
  installGlobals();
});
```

After which I get this error message:

![Screenshot 2024-01-03 at 1 52 05 PM](https://github.com/remix-run/remix/assets/13987197/305345f6-2807-4a86-8a5f-13a0d254f993)

Any ideas what I'm missing here?

**Comment by kiliman** [comment]
Do you have a sample repo?

Here's how I setup Vitest in my template.

https://github.com/kiliman/remix-vite-template/blob/main/docs/setup/vitest.md
https://github.com/kiliman/remix-vite-template/tree/main/tests/setup

And here's a sample test:
```ts
import { createRemixStub } from '@remix-run/testing'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { expect, test } from 'vitest'
import Counter, { action, loader } from './counter'

test('counter increments when clicked', async () => {
  const App = createRemixStub([
    {
      path: '/counter',
      Component: Counter,
      loader,
      action,
    },
  ])
  await render(<App initialEntries={['/counter']} />)
  const button = await screen.findByRole('button', { name: /count:/i })
  expect(button).toHaveTextContent('Count: 0')
  await userEvent.click(button)
  expect(button).toHaveTextContent('Count: 1')
})
```

**Comment by lukasalvarezdev** [comment]
I created one @kiliman https://github.com/lukasalvarezdev/jest-remix-repro

It happens with jest but not with vitest. Thanks for the help!
